### PR TITLE
Fix: Filter invalid block IDs in getBlocks() to prevent DOM duplicates

### LINKED ISSUE #7858: getBlocks() does not filter invalid block IDs (null, undefined, 'undefined')
**Describe the bug**
The `getBlocks()` function does not filter invalid block IDs (null, undefined, or the string "undefined") from `blocks_layout.items`. This can cause duplicate elements to be rendered in the DOM when invalid IDs exist, leading to test failures and potential UI issues.

**To Reproduce**

1. Make sure you are on the main branch.
2. Add the following test to `src/helpers/Blocks/Blocks.test.js` in the `describe('getBlocks')` section:
```
it('filters out invalid block IDs (null, undefined, string "undefined")', () => {
     const validBlock = { '@type': 'search', value: 'test' };
     const result = getBlocks({
       blocks: {
         'valid-id-123': validBlock,
         [null]: { '@type': 'invalid' },
         [undefined]: { '@type': 'invalid' },
         undefined: { '@type': 'invalid' },
       },
       blocks_layout: {
         items: [
           'valid-id-123',
           null,        // Invalid: null ID
           undefined,   // Invalid: undefined ID
           'undefined', // Invalid: string "undefined"
           'missing-block', // Valid ID but block doesn't exist
         ],
       },
     });
     // Should only return the valid block
     expect(result).toStrictEqual([['valid-id-123', validBlock]]);
     expect(result.length).toBe(1);
   });
```
3. Go to `cd volto/packages/volto`
4. Run the test `pnpm test src/helpers/Blocks/Blocks.test.js -t "filters out invalid block IDs"`
5. Test FAILS (proving invalid IDs are not filtered)

**Note** => Manual UI reproduction is difficult because invalid IDs rarely get into `blocks_layout.items `during normal operations (block operations use valid UUIDs), and the bug requires a very specific edge case that is hard to trigger manually. The unit test above provides a reliable way to demonstrate the issue.

**Expected behavior**
Invalid block IDs (null, undefined, 'undefined') should be filtered out before processing, so only valid block IDs are returned by `getBlocks()`.

**Screenshots**

https://github.com/user-attachments/assets/ecdf145d-9f0d-4923-9dd4-34a39b5256f4

**Software (please complete the following information):**

- OS: macOS Tahoe
- Browser Chrome
- Volto Version 19.0.0-alpha.4
- Plone Version 6.1.0
- Plone REST API Version 9.1.0

### PR DESCRIPTION
**Pull request by aryan7081**
Fixes #7858

When `blocks_layout.items` contained invalid IDs (null, undefined, 'undefined'), `getBlocks()` mapped them to `blocks[undefined]`, creating duplicate DOM elements and Cypress test failures.

This happened because `getBlocks()` didn't validate block IDs before mapping.

This fix adds `isValidBlockId()` helper to filter invalid IDs in `getBlocks()`, updates `RenderBlocks` to use it, and adds unit tests.

### CODE REVIEW

File: packages/volto/src/helpers/Blocks/Blocks.js [hunk 1]
```diff
@@ -89,11 +97,13 @@ export function blockHasValue(data) {
...
-      ?.map((n) => [n, properties?.[blocksFieldName]?.[n]])
-      .filter(([, block]) => block !== undefined) || []
-  );
+  const blocks = properties?.[blocksFieldName];
+  const items = properties?.[blocksLayoutFieldname]?.items;
+  if (!items) return [];
+  return items
+    .filter((n) => isValidBlockId(n))
+    .map((n) => [n, blocks?.[n]])
+    .filter(([, block]) => block !== undefined && block !== null);
```
Comment (wesleybl) [inline packages/volto/src/helpers/Blocks/Blocks.js]: Use `!=`

```suggestion
    .filter(([, block]) => block != null);
```

**Review by wesleybl — APPROVED** [review]

**Comment by wesleybl** [comment]
@aryan7081 Could you please do a backport to Volto 18?

**Comment by aryan7081** [comment]
@wesleybl Here is the backport PR https://github.com/plone/volto/pull/7875
Thank you!

**Comment by aryan7081** [comment]
@wesleybl on the `18.x.x` backport #7875 this breaks the `Removes invalid blocks on saving` test in `BlocksForm.test.jsx`. I’m fixing it there by having BlocksForm clean invalid layout items from the raw layout instead of relying on `getBlocks().`

I have also opened PR https://github.com/plone/volto/pull/7877 against the main branch for this fix.
